SYCL/ESIMD/api/functional/common.hpp

```diff
@@ -55,6 +55,24 @@ using shared_allocator = sycl::usm_allocator<DataT, sycl::usm::alloc::shared>;
 template <typename DataT>
 using shared_vector = std::vector<DataT, shared_allocator<DataT>>;
 
+template <typename T>
+constexpr bool should_run_test_with_current_datatype(sycl::device &&device) {
+  if constexpr (std::is_same_v<T, sycl::half>) {
+    if (!device.has(sycl::aspect::fp16)) {
+      log::note(
+          "Device does not support half precision floating point operations");
+      return false;
+    }
+  } else if constexpr (std::is_same_v<T, double>) {
+    if (!device.has(sycl::aspect::fp64)) {
+      log::note(
+          "Device does not support double precision floating point operations");
+      return false;
+    }
+  }
+  return true;
```

Comment:
```suggestion
  } else {
    // Suppress compilation warnings
    static_cast<void>(device);
    return true;
  }
```
BTW, what do you think about making the function `inline`?

Reply:
this comment was applied [here](https://github.com/intel/llvm-test-suite/pull/839/commits/1c069b5f4a52017db43d26010feca6a451457ef9#diff-ef0766a5c874c909fd25e4eec71893c4614c91112485144e4efcdab4d856d0a0R84-R86)